### ISSUE TITLE
OpenGL coordinate System

### ISSUE THREAD

**Issue by DelinQu** [HengyiWang/Co-SLAM#32]
Hi HengyiWang,

Thanks so much for releasing your code. I'm currently confused about the coordinate system of coslam in [dataloader](https://github.com/HengyiWang/Co-SLAM/blob/2329d09cc27133e710cf5e9ea1d9aaa4fa2b6662/datasets/dataset.py#L710) and [ray direction](https://github.com/HengyiWang/Co-SLAM/blob/2329d09cc27133e710cf5e9ea1d9aaa4fa2b6662/datasets/utils.py#L24). Assuming a ground truth trajectory and the corresponding mesh. The trajectory poses are flipped (only y and z) by the data loader, but the ground truth mesh stays in the original coordinate system. Will It cause any problems in the following evaluation phase? Besides, If I export a ground truth trajectory and mesh from other modeling software like Blende ..,   is it necessary to convert to openGL in both [dataloader](https://github.com/HengyiWang/Co-SLAM/blob/2329d09cc27133e710cf5e9ea1d9aaa4fa2b6662/datasets/dataset.py#L710) and [ray direction](https://github.com/HengyiWang/Co-SLAM/blob/2329d09cc27133e710cf5e9ea1d9aaa4fa2b6662/datasets/utils.py#L24)?

```python
# convert poses to openGL 
c2w[:3, 1] *= -1 # * y-axis flip
c2w[:3, 2] *= -1 # * z-axis flip
c2w[:3, 3] *= self.sc_factor

# convert rays to OpenGL
if type is  'OpenGL':
    dirs = torch.stack([(i - cx)/fx, -(j - cy)/fy, -torch.ones_like(i)], -1)
elif type is 'OpenCV':
    dirs = torch.stack([(i - cx)/fx, (j - cy)/fy, torch.ones_like(i)], -1)
```

**Comment by HengyiWang** [comment]
Hi @DelinQu, the coordinate system will not affect the reconstruction or pose evaluation. We use the OpenGL coordinate system in the code. Since ScanNet uses the OpenCV coordinate system, we need to manually convert the poses to OpenGL (y-axis flip + z-axis flip). If you want to evaluate the ground-truth mesh in this case, you can simply apply a transformation of GT mesh/reconstruction. Or you can convert the coordinate system to OpenCV in the code.
